controllers/policies/policyserver_controller.go

```diff
@@ -137,21 +137,22 @@ func (r *PolicyServerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return fmt.Errorf("failed enrolling controller with manager: %w", err)
 	}
 
-	err = mgr.GetFieldIndexer().IndexField(context.Background(), &policiesv1alpha2.PolicyServer{}, constants.PolicyServerIndexName, func(object client.Object) []string {
-		policyServer, ok := object.(*policiesv1alpha2.PolicyServer)
-		if !ok {
-			r.Log.Error(nil, "object is not type of PolicyServer: %#v", policyServer)
-			return []string{}
-		}
-		return []string{policyServer.Name}
-	})
-	if err != nil {
-		return fmt.Errorf("failed enrolling controller with manager: %w", err)
-	}
-
 	err = ctrl.NewControllerManagedBy(mgr).
 		For(&policiesv1alpha2.PolicyServer{}).
-		WithEventFilter(predicate.GenerationChangedPredicate{}).
+		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, handler.EnqueueRequestsFromMapFunc(func(object client.Object) []reconcile.Request {
```

Comment:
why do we need to watch all configmaps here? aren't we also whatching configmaps in https://github.com/kubewarden/kubewarden-controller/pull/175/files#diff-df4fe66f58973210cbb2e2c9da25f7b0c4a39d8a158043ff059763b5ab5dea23R128 ?
This will also be triggered for all configmap in the system, is it possible to at least just watch configmap for the policy-server ns?

Reply:
This controller is for `PolicyServer` resources while the one you link reconciles `ClusterAdmissionPolicy` resources.

When a configmap of a PolicyServer changes two things can happen:
- The policy server reconciler (this one), must restart the deployment. Thus, it must watch the configmap in order to force a new rollout of the deployment.
- An undetermined number of `ClusterAdmissionPolicy` status fields can become outdated (e.g. a `policyMode` changed from `monitor` to `protect`). So, it has to also watch the configmap in order to update the status of all the policies configured in the policy server. Ultimately, this is what really defines the "status" of the policy, and the source of truth for the policy `status`.